### DISCUSSION TITLE
quarkus.hibernate-orm.database.generation=update with schemes

### DISCUSSION THREAD

**Discussion by Paul6552** [quarkusio/quarkus#37924] [Q&A] [answered]
I have several projects with Quarkus and in all of them I use `quarkus.hibernate-orm.database.generation=update`. A wonderful feature.

However, I have a project where not all data is in the "public-schema" but everything is in schemes. These schemes are generated with random numbers and the number is increasing every year:
![image](https://github.com/quarkusio/quarkus/assets/25934591/dfba41c6-be88-452b-b73c-f50e5d373443)

Is there somehow a way for Quarkus to take care of all the schemes?

Currently, when we change an entity, we have to do this manually in all schemes using SQL scripts.

[a comment by an automated account is not shown]

**Comment by yrodiere** [comment, marked as answer]
Hello,

First, `quarkus.hibernate-orm.database.generation=update` [is not intended for production](https://docs.jboss.org/hibernate/orm/6.4/userguide/html_single/Hibernate_User_Guide.html#best-practices-schema). Use at your own risks. You should use Flyway/Liquibase instead.

Regarding the use of a different schema every year, you can set the default schema using [`quarkus.hibernate-orm.database.default-schema`](https://quarkus.io/guides/hibernate-orm#quarkus-hibernate-orm_quarkus.hibernate-orm.database.default-schema). The schema generation feature should take that into account.
If you need to migrate data from one schema to the next, however, that'll have to be done manually (or, again, use Flyway/Liquibase).

**Comment by doogle-oss** [comment]
`quarkus.hibernate-orm.schema-management.strategy=update`
This seems to be the updated property now.
